### PR TITLE
DAC REV B PCB Review

### PR DESCRIPTION
**Pull request by BrandtBuchda**
This PR is to approve the PCB of the DAC expansion board REV B and merge with develop. The specifications for the DAC can be found [here](https://github.com/Severson-Group/AMDC-Hardware/pull/130).

## Schematic
The final version of the schematic is located [here](https://github.com/Severson-Group/AMDC-Hardware/blob/dac-revb-pcb/Accessories/ExpansionBoard_DAC/REV20200720B/DAC_Expansion_Board_Project.pdf).

## PCB
Top layer copper routing:
image

Bottom layer copper routing:
image

3D view:
image

### Bill of Materials
The parts cost $60 per board at order quantity 10 from Digi-Key. The BOM can be viewed [here](https://github.com/Severson-Group/AMDC-Hardware/blob/dac-revb-pcb/Accessories/ExpansionBoard_DAC/REV20200720B/DAC_bom_qty_10.md)

### CODE REVIEW

**Review by npetersen2 — CHANGES_REQUESTED** [review]
@BrandtBuchda a few things before the board is ordered:

1. We need the board outline on a gerber layer from Altium. Please put it on `Mech Layer 1` by using the "Board Outline to Primitives" tool. Make the width of the outline 5mil (width doesn't technically matter, but it will get added into the board size calculation in Altium. This is okay since it *isn't* added by the fab house calculations. You could even do 1mil width). Make sure to include `Mech Layer 1` in your gerber outputs.
2. Check that [oshpark.com](https://oshpark.com/) can render the ZIP of the gerbers with no errors and it looks like you want (check silkscreen layers, mirroring, holes, etc).
3. Update the 3D PCB screenshot file in the `REVxxxx` output folder -- it looks to be old.
4. Create a cart on Digi-Key for the BOM. Let's order parts for 3 boards + 20% extras. After you make the cart, click the "Share Cart" button and send me the link. Ensure all parts are in stock and available. If something is unavailable, find an appropriate replacement part.

Ping me once these steps are complete.

**Comment by npetersen2** [comment]
@BrandtBuchda since we have ordered the boards, let's get this PR merged in ASAP... We need to get the `docs/` folder in a good state. Please try to get this PR ready to merge this week.

**Comment by elsevers** [comment]
> 
> 
> @BrandtBuchda since we have ordered the boards, let's get this PR merged in ASAP... We need to get the `docs/` folder in a good state. Please try to get this PR ready to merge this week.

I'll second this. Our new policy is to get board PRs approved before ordering. So let's get this one wrapped up before we get boards built.

**Review by npetersen2 — CHANGES_REQUESTED** [review]
@BrandtBuchda This PR is nearly ready, nice work. A few last things:
1. I see that your latest commits change the gerber files... Since we have already ordered the boards, please revert these changes so that the `REVxxxx` folder matches exactly what was ordered. Same goes with altium files.
2. This isn't an issue now, but once this is merged into `develop`, we'll need to update the links in the documentation -- they currently point to the files at a specific commit in git, but we'll want them pointing at the latest `develop` version.
3. Can you update the wordage in the `PCB Tests` file to indicate that the power supply is replacing the AMDC power supply, so when testing the board initially, do not plug it into the AMDC!
4. *Fantastic* documentation of the hardware design rationale and implementation. Your markdown doc is very easy to read and hits on all the key points of WHY your board is the way it is. Very nice.

**Review by elsevers — CHANGES_REQUESTED** [review]
Revisions pertaining to `doc/README.md`

- [ ] Revise block diagram (show isolation, add bracket for `stage 1`, add arrow for legend)
- [ ] Add content explaining jumpers
- [ ] Add photograph of board

File: Accessories/ExpansionBoard_DAC/docs/README.md [hunk 1]
```diff
@@ -1,7 +1,125 @@
...
+<img src="images/amdc-dac-jmp1.jpg" style="max-width:580px">
+
+<br>
+<br>
+
+<img src="images/amdc-dac-jmp2.jpg" style="max-width:580px">
+
+### DAC
+
+The [DAC60508MC](../REV20200720B/datasheets/dac60508.pdf) 12-bit DAC was selected for its minimal SPI interface. The CS, SCLK, and MOSI lines (supplied by the AMDC) are all that's required to drive each of the eight channels' outputs. The DAC is supplied at 5V with an internally generated _V_<sub>_REF_</sub> = 2.5V. On powerup, the DAC resets all channels' outputs to midscale which scales to 0V across the op amps. A powerup reset event can also be initiated asynchronously of the SPI interface via the onboard push button.
```
Comment (mimayberry) [inline Accessories/ExpansionBoard_DAC/docs/README.md]:
```suggestion
The [DAC60508MC](../REV20200720B/datasheets/dac60508.pdf) 12-bit DAC was selected for its minimal SPI interface. The CS, SCLK, and MOSI lines (supplied by the AMDC) are all that's required to drive each of the eight output channels. The DAC is supplied at 5V with an internally generated _V_<sub>_REF_</sub> = 2.5V. On powerup, the DAC resets all channels' outputs to midscale which scales to 0V across the op amps. A powerup reset event can also be initiated independently of the SPI interface via the onboard push button.

```

**Review by mimayberry — DISMISSED** [review]
@BrandtBuchda Very nice work, this PR looks great to me. A couple real small comments:

- I would consider explicitly spelling out that this is a digital to analog converter. I think everyone in the group would recognize what a DAC is but still.
- There are a few spots where capitalization isn't consistent, specifically when referring to the power stack and op amps. I thinks it's fine if you capitalize both words but I would be consistent each time you use them. I would also consider putting '(inverter)' after the first time you say power stack.
- The jumper section looks good, though you may consider adding a photo of each jumper configuration, e.g., "If you want to use the inverter outputs do this...[photo]", "If you want the IsoSPI do this...[photo]". I'll let the others pitch in on if this is necessary. 
- I made a few minor suggestions on wording in the `README.MD` directly.
Other than that I'm happy with this and think it's ready to be merged. Great work!

**Comment by elsevers** [comment]
@BrandtBuchda Have you completed all items requested by @npetersen2 and me and are ready for a final review? If so, can you please "re-request" our reviews?

**Review by elsevers — CHANGES_REQUESTED** [review]
Looks great! I have a few change requests related to the newly added images in `docs/README.md`

- [x] Move main image to appear lower in the document. Let's keep the document's purpose statement as the first item.

- [x] Replaces pictures with crisper photographs (these images look like the camera lens may have been dirty)

- [x] Remove photographs' background to give them a cleaner appearance (like the AMDC pictures)

Otherwise, looks great. I like that you have down sized the images to a reasonable file size.

**Comment by elsevers** [comment]
@BrandtBuchda Please re-request our review when you have implemented these changes.

**Review by npetersen2 — APPROVED** [review]
Looks good to me!

**Review by elsevers — APPROVED** [review]

**Pull request merged**